File: app/src/main/java/com/example/ontime/MyOrderAdapter.java
package com.example.ontime;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Filter;
import android.widget.Filterable;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;
import java.util.List;

public class MyOrderAdapter extends RecyclerView.Adapter<MyOrderAdapter.MyOrderViewHolder> {
    private ArrayList<MyOrderItem> mMyOrderList;
    private List<MyOrderItem> mMyOrderListFull;
    private OnItemClickListener mListener;

    public interface OnItemClickListener {
        void onItemClick(int position);
    }


    public void setOnItemClickListener(OnItemClickListener listener){
        mListener = listener;
    }

    public static class MyOrderViewHolder extends RecyclerView.ViewHolder {

        public ImageView mImageView;
        public TextView mTextView1;
        public TextView mTextView2;
        public TextView mTextView3;
        public TextView mTextView4;

        public MyOrderViewHolder(@NonNull View itemView, final OnItemClickListener listener) {
            super(itemView);
            mImageView = itemView.findViewById(R.id.my_orders_item_img);
            mTextView1 = itemView.findViewById(R.id.my_orders_rest_name);
            mTextView2 = itemView.findViewById(R.id.my_orders_order);
            mTextView3 = itemView.findViewById(R.id.my_orders_status);
            mTextView4 = itemView.findViewById(R.id.my_orders_quantity);

        }
    }

    public MyOrderAdapter(ArrayList<MyOrderItem> myOrderList) {
        mMyOrderList = myOrderList;
        mMyOrderListFull = new ArrayList<>(myOrderList);
    }

    @NonNull
    @Override
    public MyOrderViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View v = LayoutInflater.from(parent.getContext()).inflate(R.layout.my_order_item, parent, false);
        MyOrderViewHolder rvh = new MyOrderViewHolder(v, mListener);
        return rvh;
    }

    @Override
    public void onBindViewHolder(@NonNull MyOrderViewHolder holder, int position) {
        MyOrderItem currentItem = mMyOrderList.get(position);

        holder.mImageView.setImageResource(currentItem.getRestoranImage());
        holder.mTextView1.setText(currentItem.getRestoranName());
        holder.mTextView2.setText(currentItem.getOrder());
        holder.mTextView3.setText(currentItem.getStatus());
        holder.mTextView4.setText(currentItem.getQuantity());
    }

    @Override
    public int getItemCount() {
        return mMyOrderList.size();
    }


}
/*comment*/

File: app/src/main/java/com/example/ontime/MyOrderAdapter.kt
package com.example.ontime

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import java.util.*

class MyOrderAdapter(myOrderList: ArrayList<MyOrderItem>) : RecyclerView.Adapter<MyOrderAdapter.MyOrderViewHolder>() {
    private val mMyOrderList: ArrayList<MyOrderItem>
    private val mMyOrderListFull: List<MyOrderItem>
    private var mListener: OnItemClickListener? = null

    interface OnItemClickListener {
        fun onItemClick(position: Int)
    }

    fun setOnItemClickListener(listener: OnItemClickListener?) {
        mListener = listener
    }

    class MyOrderViewHolder(itemView: View, listener: OnItemClickListener?) : RecyclerView.ViewHolder(itemView) {
        var mImageView: ImageView
        var mTextView1: TextView
        var mTextView2: TextView
        var mTextView3: TextView
        var mTextView4: TextView

        init {
            mImageView = itemView.findViewById(R.id.my_orders_item_img)
            mTextView1 = itemView.findViewById(R.id.my_orders_rest_name)
            mTextView2 = itemView.findViewById(R.id.my_orders_order)
            mTextView3 = itemView.findViewById(R.id.my_orders_status)
            mTextView4 = itemView.findViewById(R.id.my_orders_quantity)
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyOrderViewHolder {
        val v = LayoutInflater.from(parent.context).inflate(R.layout.my_order_item, parent, false)
        return MyOrderViewHolder(v, mListener)
    }

    override fun onBindViewHolder(holder: MyOrderViewHolder, position: Int) {
        val currentItem: MyOrderItem = mMyOrderList[position]
        holder.mImageView.setImageResource(currentItem.restoranImage)
        holder.mTextView1.setText(currentItem.restoranName)
        holder.mTextView2.setText(currentItem.order)
        holder.mTextView3.setText(currentItem.status)
        holder.mTextView4.setText(currentItem.quantity)
    }

    override fun getItemCount(): Int {
        return mMyOrderList.size
    }

    init {
        mMyOrderList = myOrderList
        mMyOrderListFull = ArrayList<MyOrderItem>(myOrderList)
    }
}